### PR TITLE
processor, scheduler(ticdc): report resolvedTs when removing table

### LINKED ISSUE #9080: TiCDC panic: checkpointTs is greater than resolvedTs
### What did you do?

run cdc changefeed

### What did you expect to see?

no panic

### What did you see instead?

```
[root@bogon bigCluster]# kubectl --kubeconfig kubeconfig.yml -n cdc-kafka-big-cluster-tps-1712340-1-428 logs -p tc-ticdc-1
[WARN] TiCDC server data-dir is not set. Please use `cdc server --data-dir` to start the cdc server if possible.
panic: checkpointTs is greater than resolvedTs

goroutine 743 [running]:
go.uber.org/zap/zapcore.CheckWriteAction.OnWrite(0x2?, 0x2?, {0x0?, 0x0?, 0xc205230120?})
        go.uber.org/zap@v1.24.0/zapcore/entry.go:198 +0x65
go.uber.org/zap/zapcore.(*CheckedEntry).Write(0xc17e838410, {0xc23891d800, 0x2, 0x2})
        go.uber.org/zap@v1.24.0/zapcore/entry.go:264 +0x3ec
go.uber.org/zap.(*Logger).Panic(0xc000680c00?, {0x402b215?, 0x0?}, {0xc23891d800, 0x2, 0x2})
        go.uber.org/zap@v1.24.0/logger.go:258 +0x59
github.com/pingcap/log.Panic({0x402b215?, 0xc319be1d70?}, {0xc23891d800?, 0x7bec120?, 0x0?})
        github.com/pingcap/log@v1.1.1-0.20230317032135-a0d097d16e22/global.go:54 +0x8b
github.com/pingcap/tiflow/cdc/owner.(*changefeed).updateStatus(0xc060535340, 0x6215ce22ae8001a, 0x6215ce22ae80019, 0x6215ce22d40000a)
        github.com/pingcap/tiflow/cdc/owner/changefeed.go:1000 +0x192
github.com/pingcap/tiflow/cdc/owner.(*changefeed).tick(0xc060535340, {0x49668e0?, 0xc0fec17668}, 0x0?)
        github.com/pingcap/tiflow/cdc/owner/changefeed.go:500 +0x1f05
github.com/pingcap/tiflow/cdc/owner.(*changefeed).Tick(0xc060535340, {0x49667f0, 0xc0fec17638}, 0xc014826840?)
        github.com/pingcap/tiflow/cdc/owner/changefeed.go:239 +0x2d7
github.com/pingcap/tiflow/cdc/owner.(*ownerImpl).Tick(0xc067d4abd0, {0x7f03fa9b9a30?, 0xc067d49f60}, {0x4936960?, 0xc014826840})
        github.com/pingcap/tiflow/cdc/owner/owner.go:201 +0x46b
github.com/pingcap/tiflow/pkg/orchestrator.(*EtcdWorker).Run(0xc047d8f760, {0x7f03fa9b9a30, 0xc067d49f60}, 0xc0018f39e0, 0x2faf080, {0x3f7e53a, 0x5})
        github.com/pingcap/tiflow/pkg/orchestrator/etcd_worker.go:290 +0xfa6
github.com/pingcap/tiflow/cdc/capture.(*captureImpl).runEtcdWorker(0xc000536c40, {0x4966890?, 0xc067d49f60}, {0x7f03eea5af58, 0xc067d4abd0}, {0x4936960, 0xc014826840}, 0x0?, {0x3f7e53a, 0x5})
        github.com/pingcap/tiflow/cdc/capture/capture.go:522 +0x185
github.com/pingcap/tiflow/cdc/capture.(*captureImpl).campaignOwner(0xc000536c40, {0x4966890?, 0xc000d7e6c0})
        github.com/pingcap/tiflow/cdc/capture/capture.go:473 +0x814
github.com/pingcap/tiflow/cdc/capture.(*captureImpl).run.func3()
        github.com/pingcap/tiflow/cdc/capture/capture.go:328 +0x3c
golang.org/x/sync/errgroup.(*Group).Go.func1()
        golang.org/x/sync@v0.2.0/errgroup/errgroup.go:75 +0x64
created by golang.org/x/sync/errgroup.(*Group).Go
        golang.org/x/sync@v0.2.0/errgroup/errgroup.go:72 +0xa5
```

### Versions of the cluster

TiCDC version (execute `cdc version`):

```console
Release Version: v7.1.0-test
Git Commit Hash: 49faafdaa53d54ab817845af7517f90a8fe10623
Git Branch: heads/refs/tags/v7.1.0-test
UTC Build Time: 2023-05-26 06:53:18
Go Version: go version go1.20.3 linux/amd64
Failpoint Build: false
```

### PR DESCRIPTION
**Pull request by CharlesCheung96**
### What problem does this PR solve?

Issue Number: close #9080

### What is changed and how it works?
 report resolvedTs when removing table 

#### Tests 

 - Unit test
 - Integration test

### Release note 

```release-note
`None`.
```

### CODE REVIEW

File: cdc/scheduler/internal/v3/replication/replication_set.go [hunk 1]
```diff
@@ -915,6 +915,10 @@ func (r *ReplicationSet) handleCaptureShutdown(
 func (r *ReplicationSet) updateCheckpointAndStats(
 	checkpoint tablepb.Checkpoint, stats tablepb.Stats,
 ) {
+	if checkpoint.ResolvedTs < checkpoint.CheckpointTs {
+		log.Panic("schedulerv3: resolved ts must not be greater than checkpoint ts",
```
Comment (asddongmen) [inline cdc/scheduler/internal/v3/replication/replication_set.go:919]:
```suggestion
		log.Panic("schedulerv3: resolved ts must net be less than checkpoint ts",
```

File: cdc/scheduler/internal/v3/agent/table.go [hunk 2]
```diff
@@ -79,8 +79,11 @@ func newAddTableResponseMessage(status tablepb.TableStatus) *schedulepb.Message
 		DispatchTableResponse: &schedulepb.DispatchTableResponse{
 			Response: &schedulepb.DispatchTableResponse_AddTable{
 				AddTable: &schedulepb.AddTableResponse{
-					Status:     &status,
-					Checkpoint: status.Checkpoint,
+					Status: &status,
+					Checkpoint: tablepb.Checkpoint{
```
Comment (asddongmen) [inline cdc/scheduler/internal/v3/agent/table.go:83]: Should it be `newRemoveTableResponseMessage` ?

File: cdc/scheduler/internal/v3/agent/table.go [hunk 3]
```diff
@@ -79,8 +79,11 @@ func newAddTableResponseMessage(status tablepb.TableStatus) *schedulepb.Message
...
+						ResolvedTs:   status.Checkpoint.CheckpointTs,
```
Comment (nongfushanquan) [inline cdc/scheduler/internal/v3/agent/table.go:84]: why is the resolved ts assigned with the value  checkpoints

File: cdc/scheduler/internal/v3/agent/table.go [hunk 4]
```diff
@@ -88,6 +93,15 @@ func newAddTableResponseMessage(status tablepb.TableStatus) *schedulepb.Message
 }
 
 func newRemoveTableResponseMessage(status tablepb.TableStatus) *schedulepb.Message {
+	if status.Checkpoint.ResolvedTs < status.Checkpoint.CheckpointTs {
```
Comment (asddongmen) [inline cdc/scheduler/internal/v3/agent/table.go:96]: ResolvedTs is 0 here, it always panics.

**Comment by asddongmen** [comment]
Please add unit test to cover this case after the PR.

File: cdc/scheduler/internal/v3/agent/table.go [hunk 5]
```diff
@@ -74,6 +74,11 @@ func (t *tableSpan) getTableSpanStatus(collectStat bool) tablepb.TableStatus {
 }
 
 func newAddTableResponseMessage(status tablepb.TableStatus) *schedulepb.Message {
+	if status.Checkpoint.ResolvedTs < status.Checkpoint.CheckpointTs {
```
Comment (3AceShowHand) [inline cdc/scheduler/internal/v3/agent/table.go:77]: also print table id, or other useful debug information.

**Review by 3AceShowHand — APPROVED** [review]

File: cdc/scheduler/internal/v3/agent/table.go (hunk 3 above)
Comment (CharlesCheung96) [reply cdc/scheduler/internal/v3/agent/table.go]: Because resolvedTs should equal to checkpointTs when removing table.

**Review by asddongmen — APPROVED** [review]

**Pull request merged**